processor/schemaprocessor/internal/context/optimised.go

```diff
@@ -0,0 +1,63 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package context
+
+import "context"
+
+type (
+	Context    = context.Context
+	CancelFunc = context.CancelFunc
+
+	// optimized is a lock free implementation of context.Context.
```

Comment:
This is probably not necessary at all for this processor, but it would be neat to see if it would help speed up processing in the collector in general.